tests/unit/test_messages.py

```
@@ -396,6 +396,70 @@ async def mock_stream() -> Any:
+async def test_default_amessages_streaming_usage_from_trailing_chunk() -> None:
+    """Usage (incl. cache) from a trailing usage-only chunk after finish_reason is reported in message_delta."""
+    from any_llm.types.completion import (
+        ChatCompletionChunk,
+        ChoiceDelta,
+        ChunkChoice,
+        CompletionUsage,
+        PromptTokensDetails,
+    )
+    from any_llm.types.messages import MessageDeltaEvent, MessagesParams
+
+    async def mock_stream() -> Any:
```

Ruff (0.15.18)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `mock_stream`

(ANN401)